File: forest_contest/03/03-004.rb
# キーボードのシミュレーション (paizaランク C 相当)
# https://paiza.jp/works/mondai/forest_contest_003/forest_contest_003__c_keyboard

# 入出力例
INPUT1 = <<"EOS"
3
a
shift b
c
EOS
OUTPUT1 = <<"EOS"
aBc
EOS

INPUT2 = <<"EOS"
7
a
capslock
b
c
d
capslock
f
EOS
OUTPUT2 = <<~"EOS"
  aBCDf
EOS

# 解答例1
# 入力
n = gets.to_i
c = n.times.map { gets.split }

display = ""
capslock = false
# n 行の入力で繰り返し処理
c.each do |keys|
  key1, key2 = keys
  case key1
  when "capslock"
    # capslock の ON/OFF 切り替え
    capslock = !capslock
  when "shift"
    # shift と一緒に押されたキーの大文字を display に追加
    display <<= key2.upcase
  else
    # capslock ON なら大文字, OFF なら小文字を display に追加
    display <<= capslock ? key1.upcase : key1
  end
end

# 出力
puts display

File: forest_contest/03/03-004.py
# キーボードのシミュレーション (paizaランク C 相当)
# https://paiza.jp/works/mondai/forest_contest_003/forest_contest_003__c_keyboard

# 入出力例
INPUT1 = """\
3
a
shift b
c"""
OUTPUT1 = """\
aBc"""

INPUT2 = """\
7
a
capslock
b
c
d
capslock
f"""
OUTPUT2 = """\
aBCDf"""

# python3 解答例
# 入力
n = int(input())
c = [input() for _ in range(n)]

display = ""
capslock = False
# n 行の入力で繰り返し処理
for row in c:
    keys = row.split()

    if keys[0] == "capslock":
        # capslock の ON/OFF 切り替え
        capslock = not capslock
    elif keys[0] == "shift":
        # shift と一緒に押されたキーの大文字を display に追加
        display += keys[1].upper()
    else:
        # capslock ON なら大文字, OFF なら小文字を display に追加
        display += keys[0].upper() if capslock else keys[0]

# 出力
print(display)
